### ISSUE TITLE
Failing to recreate GoogleApiClient after upgrading play services 9.0.2

### ISSUE THREAD

**Issue by vitovalov** [googlesamples/android-fit#18]
I made a [fork](https://github.com/vitovalov/android-fit) from android-fit google's github project and added a couple of commits. 

The problem comes with play-services-fitness version upgrade. It works as expected in 8.4.0 but if you upgrade to 9.0.2 appears the issue of recreating googleApiClient. 

I'm trying to solve the case when the googleApiClient has to be recreated because stopAutoManage has been automatically called due to some error (either user canceled the sign-in dialog, or other client was alive but was manually stopAutomanage called). See the gifs.

So to sum up, the question is how to properly recreate the GoogleApiClient with play services 9+?

There's also an [issue](https://code.google.com/p/android/issues/detail?id=218157) in android bug tracker.

**Comment by codingjeremy** [comment]
I am closing this issue/PR, as it has been migrated to the new repo linked above in the comments. Thank you!
